### ISSUE TITLE
[Bug]: Proton not able to run apps with WineD3D, missing dependencies

### ISSUE THREAD

**Issue by SeongGino** [bottlesdevs/Bottles#3485]
### Describe the bug

When running a prefix using Proton instead of Wine, if an app uses WineD3D (namely anything that uses DirectDraw/`ddraw.dll`), Proton will not be able to run it due to missing two libraries: `libvkd3d-1.dll` & `libvkd3d-shader-1.dll`. The issue is only resolved by falling back to a different version of Wine rather than Proton - alternatively, symlinking/copying the files to `$PREFIX/drive_c/windows/[system32/syswow64]` resolves the issue for Proton and the application will run.

For what it's worth, the libraries are included in `{PROTON}/files/lib[64]/vkd3d`

Tested with *GE-Proton9-11* downloaded from inside Bottles, and *Proton - Experimental* symlinked from a working Steam install.

### To Reproduce

1. Create a new Prefix using a version of Proton.
2. Start a game that uses `wined3d.dll` rather than DXVK/VKD3D.
3. Observe the error attached.

### Package

Flatpak from Flathub

### Distribution

Arch Linux

### Debugging Information

```shell
Official Package: true
Version: '51.13'
DE/WM: plasma
Display:
    X.org: true
    X.org (port): :1
    Wayland: true
Graphics:
    vendors:
        nvidia:
            vendor: nvidia
            envs:
                __NV_PRIME_RENDER_OFFLOAD: '1'
                __GLX_VENDOR_LIBRARY_NAME: nvidia
                __VK_LAYER_NV_optimus: NVIDIA_only
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json
            nvngx_path: /usr/lib/x86_64-linux-gnu/GL/nvidia-560-35-03/extra/nvidia/wine
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 6.9.8-273-tkg-pds
Disk:
    Total: 8339234816
    Free: 8339066880
RAM:
    MemTotal: 15.5GiB
    MemAvailable: 9.2GiB
Bottles_envs: null
```

### Troubleshooting Logs

```shell
21:59:46 (INFO) Launching with URI: None 
21:59:46 (INFO) [Activate] request received. 
21:59:46 (INFO) Bottles Started! 
21:59:47 (INFO) Performing Bottles checks… 
21:59:47 (INFO) Dxvks found:
         - dxvk-2.4
 
21:59:47 (INFO) Vkd3ds found:
         - vkd3d-proton-2.13
 
21:59:47 (INFO) Nvapis found:
         - dxvk-nvapi-v0.7.1
 
21:59:47 (INFO) Latencyflexs found:
         - latencyflex-v0.1.1
 
21:59:47 (INFO) Runners found:
         - soda-9.0-1
         - ge-proton9-11
         - sys-wine-9.0
 
21:59:47 (INFO) Bottles found:
         - Test
 
21:59:48 (INFO) Catalog installers loaded 
21:59:48 (INFO) Catalog dependencies loaded 
21:59:48 (INFO) Catalog components loaded 
21:59:49 (INFO) Setting Key sync=fsync for bottle Test… 
21:59:49 (INFO) There is no running wineserver. 
21:59:55 (INFO) Launching an executable… 
21:59:55 (INFO) Using EasyAntiCheat runtime 
21:59:55 (INFO) Using BattlEye runtime 
wine: using kernel write watches, use_kernel_writewatch 1.
fsync: up and running.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
wine: using kernel write watches, use_kernel_writewatch 1.
00dc:err:module:import_dll Library libvkd3d-1.dll (which is needed by L"C:\\windows\\system32\\wined3d.dll") not found
00dc:err:module:import_dll Library libvkd3d-shader-1.dll (which is needed by L"C:\\windows\\system32\\wined3d.dll") not found
00dc:err:module:import_dll Library wined3d.dll (which is needed by L"C:\\windows\\system32\\DDRAW.dll") not found
00dc:err:module:import_dll Library DDRAW.dll (which is needed by L"<redacted>") not found
00dc:err:module:loader_init Importing dlls for L"<redacted>" failed, status c0000135
```

### Additional context

I've mostly noticed this with regards to apps using DirectDraw in some form, and it applies to a few different games, mainly older titles (like the 1998 PC port of Final Fantasy VII).

**Comment by keenanweaver** [comment]
This is still an issue on the latest Bottles. I don't believe I ever saw this issue before 51.13/51.12/51.11. However, I did downgrade to 51.10 for testing and still saw the problem. Attempted multiple versions of the Proton-GE runner (with and without the Steam runtime), and it still resolves the above error.

And to clarify, it doesn't need to start as a Proton bottle to experience the problem; swapping an existing Wine runner to a Proton runner will exhibit the problem. Some games I tested: NFS: Underground, The Sims: Complete Edition, LEGO Island.

I also tested a non-Flatpak version of Bottles (nix package), and the same problem was there.

My debugging info:
```
Official Package: true
Version: '51.14'
DE/WM: plasma
Display:
    X.org: true
    X.org (port): :0
    Wayland: true
Graphics:
    vendors:
        amd:
            vendor: amd
            envs:
                DRI_PRIME: '1'
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/radeon_icd.x86_64.json:/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/radeon_icd.i686.json
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 6.11.2
Disk:
    Total: 16788467712
    Free: 16788295680
RAM:
    MemTotal: 31.3GiB
    MemAvailable: 23.4GiB
Bottles_envs: null
```

**Comment by james2432** [comment]
explains why ge-proton doesn't launch applications for me but sys-wine does(No vulkan available on laptop, explicitly vulkan disabled via UI options)

**Comment by Sturmlocke86** [comment]
> This is still an issue on the latest Bottles. I don't believe I ever saw this issue before 51.13/51.12/51.11. However, I did downgrade to 51.10 for testing and still saw the problem. Attempted multiple versions of the Proton-GE runner (with and without the Steam runtime), and it still resolves the above error.
> 

@keenanweaver I'm trying to test something with the ProtonGE runner right now and I can't see the checkbox in any bottle -> settings -> compatibility to enable the additional Steam runtime. Should it just be in there under compatibility or does anything else have to be done for the option to enable / disable the runtime show up? Cheers

**Comment by SeongGino** [comment]
> I can't see the checkbox in any bottle -> settings -> compatibility to enable the additional Steam runtime.

You mean you don't see this? 
![2024_11-04 234655-Bottles](https://github.com/user-attachments/assets/e5341f66-fb9c-44a4-810d-81ea40b1b474)

**Comment by Sturmlocke86** [comment]
> > I can't see the checkbox in any bottle -> settings -> compatibility to enable the additional Steam runtime.
> 
> You mean you don't see this? ![2024_11-04 234655-Bottles](https://private-user-images.githubusercontent.com/7321839/383009645-e5341f66-fb9c-44a4-810d-81ea40b1b474.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3ODI0NjQsIm5iZiI6MTczMDc4MjE2NCwicGF0aCI6Ii83MzIxODM5LzM4MzAwOTY0NS1lNTM0MWY2Ni1mYjljLTQ0YTQtODEwZC04MWVhNDBiMWI0NzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDVUMDQ0OTI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDEzM2M1NDlkNDc0YmY2NDg0MjQ3NjcwMzM5MDdhNTBhM2ZlZTQwYTBhYzA1ODIzOTBlYmU4MTc1NTJhNjkzNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TQPVmW_hCm1z2NffhJngvrXRsKRlNDY93g_LYG2oO40)

Nope, neither the Bottles runtime nor the Steam runtime checkbox is there for me. I just installed bottles freshly on Ubuntu 24.10 and let it install ProtonGE but the two entries in your screenshot are not there. I can see the rest though.

**Comment by Sturmlocke86** [comment]
> > > I can't see the checkbox in any bottle -> settings -> compatibility to enable the additional Steam runtime.
> > 
> > 
> > You mean you don't see this?
> 
> Nope, neither the Bottles runtime nor the Steam runtime checkbox is there for me. I just installed bottles freshly on Ubuntu 24.10 and let it install ProtonGE but the two entries in your screenshot are not there. I can see the rest though.

This is how it looks for me: 
![bottles, missing entries, sturm, temp](https://github.com/user-attachments/assets/acf5b326-6e1e-422f-955f-2a267a67ae5f)

**Comment by SeongGino** [comment]
This is the current git version, so no idea why yours would be different unless it was out of date. :shrug:

**Comment by Sturmlocke86** [comment]
> This is the current git version, so no idea why yours would be different unless it was out of date. 🤷

Hm, I'm on version 51.15, which should be the newest stable build on Flathub. I'll install it on my secondary pc with Suse and check if the same thing happens here too.

**Comment by james2432** [comment]
51.15 flathub stable, archlinux:
![image](https://github.com/user-attachments/assets/27a20539-4d3c-4121-8896-7d4091c1d37f)

```
Bottles                                         com.usebottles.bottles                              51.15                              stable                 user
Freedesktop Platform                            org.freedesktop.Platform                            freedesktop-sdk-23.08.25           23.08                  system
i386                                            org.freedesktop.Platform.Compat.i386                                                   23.08                  user
Mesa                                            org.freedesktop.Platform.GL.default                 24.2.5                             23.08                  system
Mesa (Extra)                                    org.freedesktop.Platform.GL.default                 24.2.5                             23.08-extra            system
Mesa                                            org.freedesktop.Platform.GL.default                 24.2.5                             24.08                  user
Mesa (Extra)                                    org.freedesktop.Platform.GL.default                 24.2.5                             24.08extra             user
Mesa                                            org.freedesktop.Platform.GL32.default               24.2.5                             23.08                  user
Mesa                                            org.freedesktop.Platform.GL32.default               24.2.5                             24.08                  user
FFmpeg extension with extra codecs              org.freedesktop.Platform.ffmpeg-full                                                   23.08                  user
FFmpeg extension with extra codecs              org.freedesktop.Platform.ffmpeg-full                                                   24.08                  user
i386                                            org.freedesktop.Platform.ffmpeg_full.i386                                              23.08                  user
i386                                            org.freedesktop.Platform.ffmpeg_full.i386                                              24.08                  user
openh264                                        org.freedesktop.Platform.openh264                   2.1.0                              2.2.0                  system
openh264                                        org.freedesktop.Platform.openh264                   2.4.1                              2.4.1                  user
GNOME Application Platform version 47           org.gnome.Platform                                                                     47                     user
i386                                            org.gnome.Platform.Compat.i386                                                         47                     user
Adwaita dark GTK theme                          org.gtk.Gtk3theme.Adwaita-dark                                                         3.22                   system
Adwaita dark GTK theme                          org.gtk.Gtk3theme.Adwaita-dark                                                         3.22                   user
Wine                                            org.winehq.Wine                                     9.0                                stable-23.08           user
DXVK                                            org.winehq.Wine.DLLs.dxvk                           2.4.1                              stable-23.08           user
Gecko                                           org.winehq.Wine.gecko                                                                  stable-23.08           user
gecko                                           org.winehq.Wine.gecko                                                                  stable-24.08           user
Mono                                            org.winehq.Wine.mono                                                                   stable-23.08           user
mono                                            org.winehq.Wine.mono                                                                   stable-24.08           user
```

**Comment by Sturmlocke86** [comment]
I have Steam Flatpak installed in both cases (Ubuntu and openSUSE), and both have Steam runtime 3.0 sniper installed. So Bottles takes the runtime from a preinstalled Steam installation? I always thought it was bundled and included with the Bottles Flatpak (or that it would download it after enabling it via checkbox). If this is the case, might I need to give Bottles permission to access Steam Flatpak via Flatseal maybe? I ask because checking Bottles -> settings -> Steam prefix it tells me that I can't enable that feature, because it couldn't find Steam due to lacking permissions etc.

**Comment by keenanweaver** [comment]
Yeah, I may be full of it 'cause I just wiped out the Steam flatpak and directory altogether, created a new bottle, and the runtime shows up. No idea why this differs from my previous attempts. It even shows up on runners like Soda, when it didn't before. Still no Bottles runtime, though.

I [manage my permissions through NixOS](https://github.com/keenanweaver/nix-config/blob/7342382191e70eaa6c9a43a3e50bb6d700a846c3/modules/profiles/gaming.nix#L607), so no idea why it's different now.

**Comment by Sturmlocke86** [comment]
That's interesting. I checked the Bottles doc to see if it has any info on how to enable the runtimes or how it works exactly, but didn't find anything relevant. Maybe @mirkobrombin can help us out here?

**Comment by SeongGino** [comment]
Sounds like a separate issue from the one being discussed right now. If it's a problem, you should write a separate issue about it rather than hijacking this thread.

**Comment by CaptainCoward** [comment]
Same here for me.. Wanted to test some Proton-GE since i have issues with EAC on the Game "Deceive Inc." Epic just gave away. Was looking for where to enable steam runtime.. checked compability. Nothing. Just looks like https://github.com/bottlesdevs/Bottles/issues/3485#issuecomment-2457171694 Do you need to enable[ Steam Proton Prefixes](https://docs.usebottles.com/flatpak/cant-enable-steam-proton-manager) for that to appear?

**Comment by mirkobrombin** [comment]
Proton must always be used with the Steam runtime enabled. Bottles show that option only if steam is found in the common paths.

**Comment by Sturmlocke86** [comment]
> Proton must always be used with the Steam runtime enabled. Bottles show that option only if steam is found in the common paths.

Thanks for taking the time, Mirko. So do we have to grant Bottles access to the Steam Flatpak path via Flatseal or is it done any other way? Or should Bottles auto-detect Steam just by having Steam Flatpak installed?

**Comment by mirkobrombin** [comment]
> > Proton must always be used with the Steam runtime enabled. Bottles show that option only if steam is found in the common paths.
> 
> Thanks for taking the time, Mirko. So do we have to grant Bottles access to the Steam Flatpak path via Flatseal or is it done any other way? Or should Bottles auto-detect Steam just by having Steam Flatpak installed?

Correct. You have to give it access to the Steam path, according to its installation method.

**Comment by Sturmlocke86** [comment]
> > > Proton must always be used with the Steam runtime enabled. Bottles show that option only if steam is found in the common paths.
> > 
> > 
> > Thanks for taking the time, Mirko. So do we have to grant Bottles access to the Steam Flatpak path via Flatseal or is it done any other way? Or should Bottles auto-detect Steam just by having Steam Flatpak installed?
> 
> Correct. You have to give it access to the Steam path, according to its installation method.

[That worked like a charm](https://docs.usebottles.com/flatpak/cant-enable-steam-proton-manager), **thank you**! Steam runtime is available now (checkbox for the Bottles runtime is still missing though). PS: It would be nice if both could be bundled with the Bottles Flatpak for ease of use. Cheers

**Comment by Sturmlocke86** [comment]
I'm adding myself to the users affected by the issue the OP is describing. I'm trying to get an old game (Vanguard: Saga of Heroes) to launch and I'm facing the same issue as mentioned in this bug report:

`0388:err:module:import_dll Library libvkd3d-1.dll (which is needed by L"C:\\windows\\system32\\wined3d.dll") not found
0388:err:module:import_dll Library libvkd3d-shader-1.dll (which is needed by L"C:\\windows\\system32\\wined3d.dll") not found
0388:err:module:import_dll Library wined3d.dll (which is needed by L"C:\\windows\\system32\\DDRAW.dll") not found
0388:err:module:import_dll Library DDRAW.dll (which is needed by L"C:\\Vanguard EMU\\bin\\VGClient.exe") not found
0388:err:module:loader_init Importing dlls for L"C:\\Vanguard EMU\\bin\\VGClient.exe" failed, status c0000135`

The above is with DXVK enabled in the bottles setting and it makes no difference whether it's on or off and the same goes for the Steam runtime. The above is with a custom-patched ProtonGE build, needed for the game to run, but I also tried with an unmodified GE-Proton 9-20 runner. Testing with GE-Proton 7-4 made the game launch though. Syswine works as do a couple of other runners. Newer games run just fine.

**Comment by yehorb** [comment]
I encountered this problem too recently, while trying to run Crysis of all things.

I copied the relevant `libvkd3d*` files from Steam `compatdata` to bottle directory, and was able to successfully run the game:

```bash
> pwd
/home/y/.var/app/com.usebottles.bottles/data/bottles/bottles/Crysis/drive_c/windows
> cp ~/.local/share/Steam/steamapps/compatdata/0/pfx/drive_c/windows/system32/libvkd3d* system32/
> cp ~/.local/share/Steam/steamapps/compatdata/0/pfx/drive_c/windows/syswow64/libvkd3d* syswow64/
```

I think you need to have at least one game that runs via Proton in Steam so that `~/.local/share/Steam/steamapps/compatdata` exists.

**Comment by lyoneel** [comment]
I can confirm @yehorb solution works just fine, I got the same error with StarcraftII on a "gaming" based bottle using protonGE 10.32.

**Comment by fira** [comment]
Hi - Still having this same issue

I added Flatpak permissions for Bottles to be able to detect the SLR (to make it clear, you don't need to install Steam via Flatpak if you haven't - giving it permissions to `~/.local/share/Steam` using something like Flatseal is enough), but this didn't make any difference:

```
01e0:err:module:import_dll Library libvkd3d-1.dll (which is needed by L"C:\\windows\\system32\\wined3d.dll") not found
01e0:err:module:import_dll Library libvkd3d-shader-1.dll (which is needed by L"C:\\windows\\system32\\wined3d.dll") not found
01e0:err:module:import_dll Library wined3d.dll (which is needed by L"C:\\windows\\system32\\DDRAW.dll") not found
01e0:err:module:import_dll Library DDRAW.dll (which is needed by L"C:\\Games\\Homecoming\\bin\\win64\\live\\cityofheroes.exe") not found
``` 

Didn't try yehorb's fix because I'm trying to get this fixed the right way...
As far as I can tell, the files are right there alongside the Proton Runners, but when using Bottles they're not used, whereas launching on Steam works properly.

```
./runners/ge-proton10-34/files/share/default_pfx/drive_c/windows/system32/libvkd3d-1.dll
./runners/ge-proton10-34/files/share/default_pfx/drive_c/windows/system32/libvkd3d-shader-1.dll
./runners/ge-proton10-34/files/share/default_pfx/drive_c/windows/syswow64/libvkd3d-1.dll
./runners/ge-proton10-34/files/share/default_pfx/drive_c/windows/syswow64/libvkd3d-shader-1.dll
```

This is especially silly that it also affects legacy DX9 games. Right now I'm not sure what to do besides just use Steam "Non-Game" shortcuts as a poor man's prefix manager...

**Comment by mirkobrombin** [comment]
Fixed by #4595. Bottles now copies the Proton VKD3D support DLLs into the prefix before launching with WineD3D.
